shared/src/question-generators/graph-algorithms/dijkstraTable.ts

```diff
@@ -0,0 +1,440 @@
+    const prevFieldID = getPreviousFieldID(fieldID)
+    if (prevFieldID && text[prevFieldID]) {
+      const previousInput = text[prevFieldID]?.trim() || ""
+      return checkConsistency(input, previousInput)
+    }
+
+    return { valid: true, message: "" }
+  }
+}
+
+function getFeedbackFunction(
+  steps: Array<{
+    set: Set<string>
+    distances: Record<string, string>
+    predecessors: Record<string, string>
+  }>,
+  graph: Graph,
+): MultiFreeTextFeedbackFunction {
+  return ({ text }) => {
+    let allStepsCorrect = true
+
+    // generate feedback table
+    const headerRow = `| $S$ | ${graph.nodes.map((n) => `$(d(${n.label} ), p(${n.label} ))$ |`).join(" ")}`
+    const separatorRow = `| --- | ${graph.nodes.map(() => "---:|").join(" ")}`
+
+    const feedbackDetails = steps
+      .map((step, stepIndex) => {
+        // handle prefilled row
+        if (stepIndex === 0) {
+          const nodeFeedback = graph.nodes.map(() => `$( , )$`).join(" | ")
+          return `| $\\{${graph.nodes[0].label}\\}$ | ${nodeFeedback} |`
+        }
+
+        // handle user input rows
+        const setField = `step${stepIndex}_s`
+        const expectedSet = `{${[...step.set].join(", ")}}`
+        const userNewNode = text[setField]?.trim() || ""
+        const expectedNewNode = [...step.set].filter((node) => !steps[stepIndex - 1]?.set.has(node))[0]
+
+        if (userNewNode !== expectedNewNode) allStepsCorrect = false
+
+        const previousStep = steps[stepIndex - 1]
+        const rowContent = graph.nodes
+          .map((node) => {
+            const nodeLabel = node.label!
+
+            const distanceField = `step${stepIndex}_${nodeLabel}_d`
+            const userDistance = text[distanceField]?.trim().replace(/^\s*$/, "") || ""
+            const expectedDistance = step.distances[nodeLabel] || previousStep.distances[nodeLabel] || ""
+
+            const predecessorField = `step${stepIndex}_${nodeLabel}_p`
+            const userPredecessor = text[predecessorField]?.trim().replace(/^\s*$/, "") || ""
+            const expectedPredecessor =
+              step.predecessors[nodeLabel] || previousStep.predecessors[nodeLabel] || ""
+
+            if (
+              (userDistance !== "" && userDistance !== expectedDistance) ||
+              (userPredecessor !== "" && userPredecessor !== expectedPredecessor)
+            ) {
+              allStepsCorrect = false
+            }
+
+            return `$(${expectedDistance}, ${expectedPredecessor})$`
+          })
+          .join(" | ")
+
+        return `| $\\{${expectedSet}\\}$ | ${rowContent} |`
+      })
+      .join("\n")
+
+    // final table structure
+    const completeTable = `${headerRow}\n${separatorRow}\n${feedbackDetails}`
+    //console.log("Generated Feedback Table:\n", completeTable)
+
+    return {
+      correct: allStepsCorrect,
+      correctAnswer: completeTable,
+    }
+  }
+}
+
+function generateDijkstraSteps(graph: Graph, startLabel: string) {
```

Comment:
Although the output of this algorithm differs slightly from the other Dijkstra algorithm, could we possibly combine the functions? Perhaps we can modify the `interface DijkstraResult` to include order removal. This would eliminate redundant logic.